### PR TITLE
Pass processable attestations to pool

### PR DESCRIPTION
**Pull request by cemozerr**
## PR Description
Pass processable attestations to pool

### CODE REVIEW

File: sync/src/main/java/tech/pegasys/teku/sync/AttestationManager.java [hunk 1]
```diff
@@ -40,42 +41,42 @@
...
   private final FutureItems<DelayableAttestation> futureAttestations;
+  private final Consumer<Attestation> attestationPoolConsumer;
 
   AttestationManager(
       final EventBus eventBus,
       final ForkChoiceAttestationProcessor attestationProcessor,
       final PendingPool<DelayableAttestation> pendingAttestations,
-      final FutureItems<DelayableAttestation> futureAttestations) {
+      final FutureItems<DelayableAttestation> futureAttestations,
+      final Consumer<Attestation> attestationPoolConsumer) {
```
Comment (ajsutton) [inline sync/src/main/java/tech/pegasys/teku/sync/AttestationManager.java:51]: Is there a reason this doesn't just take the `AggregatingAttestationPool`? Then `AttestationManager` would really be heading towards the thing that encompasses all attestation components like I'd hoped for.

File: sync/src/main/java/tech/pegasys/teku/sync/AttestationManager.java [hunk 2]
```diff
@@ -40,42 +41,42 @@
...
   }
 
   @Subscribe
   @SuppressWarnings("unused")
   private void onAggregateAndProof(final SignedAggregateAndProof aggregateAndProof) {
     final Attestation aggregate = aggregateAndProof.getMessage().getAggregate();
-    processAttestation(
-        new DelayableAttestation(
-            aggregate, () -> eventBus.post(new ProcessedAggregateEvent(aggregate))));
+    processAttestation(new DelayableAttestation(aggregate, attestationPoolConsumer));
```
Comment (ajsutton) [inline sync/src/main/java/tech/pegasys/teku/sync/AttestationManager.java:79]: I think we still need the distinction between posting a `ProcessedAttestationEvent` or a `ProcessedAggregateEvent`, just that in both cases we'd also add to the `AggregatingAttestationManager`. Or is nothing listening for those events?  If not, how do we wind up gossiping the attestation once it has been processed if it was initially delayed?

**Review by ajsutton — COMMENTED** [review]
Generally looks like it's going in the right direction.  There may be more problems with how attestations are flowing through the system than I first thought though....

**Comment by ajsutton** [comment]
Also, if we're aggregating for a slot - wouldn't we need to add gossiped attestations to the aggregating pool before they can actually be processed (ie they are aggregated in the current slot but can't be processed until the next one).  We can't include them in a block yet, but the AggregatingAttestationPool will filter for that.

File: sync/src/main/java/tech/pegasys/teku/sync/AttestationManager.java (hunk 2 above)
Comment (cemozerr) [reply sync/src/main/java/tech/pegasys/teku/sync/AttestationManager.java:79]: Made a ticket for the attestations we should gossip after processing: https://pegasys1.atlassian.net/secure/RapidBoard.jspa?rapidView=67&modal=detail&selectedIssue=BC-450

**Comment by cemozerr** [comment]
Closing this PR in favor of a better fix.

**Pull request closed without merging**